### PR TITLE
Add limits to recovery mode.

### PR DESCRIPTION
**Pull request by jubeira**
# Description

Proportional exits don't really need limits, but they don't hurt either. They can always be just set to 0 if they are not desired.

This PR implements limits for recovery mode.

## Type of change

- [ ] Bug fix 
- [x] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

## Issue Resolution

N/A

### CODE REVIEW

File: pkg/vault/test/.contract-sizes/Router [hunk 1]
```diff
@@ -1,2 +1,2 @@
-Bytecode	24.003* (3 over)
-InitCode	25.355
\ No newline at end of file
+Bytecode	24.303* (310 over)
+InitCode	25.662
```
Comment (joaobrunoah) [inline pkg/vault/test/.contract-sizes/Router:2]: Probably splitting the buffer router will solve this, right?

File: pkg/vault/test/foundry/RecoveryMode.t.sol [hunk 2]
```diff
@@ -22,11 +22,84 @@ contract RecoveryModeTest is BaseVaultTest {
...
+                expectedAmountsOut[daiIdx] - 1,
+                expectedAmountsOut[daiIdx]
+            )
+        );
+        vm.prank(alice);
+        router.removeLiquidityRecovery(pool, amountToRemove, expectedAmountsOut);
     }
 
-    function testRecoveryModeEmitsPoolBalanceChangedEvent() public {
+    function testRecoveryModeEmitsPoolBalanceChangedEquvent() public {
```
Comment (joaobrunoah) [inline pkg/vault/test/foundry/RecoveryMode.t.sol:102]:
```suggestion
    function testRecoveryModeEmitsPoolBalanceChangedEvent() public {
```

**Review by joaobrunoah — APPROVED** [review]
LGTM!

**Review by elshan-eth — APPROVED** [review]
LGTM!

**Review by danielmkm — APPROVED** [review]
This looks good

**Pull request merged**